src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -546,5 +546,24 @@ public static bool SequenceEqual<T>(this IEnumerable<T> first, IEnumerable<T> se
 
             return true;
         }
+
+        public static T AggregateOrDefault<T>(this IEnumerable<T> source, Func<T, T, T> func)
```

Comment:
cute!

Reply:
nit: would it be simpler to assume an aggregation function that does nothing on default values, allowing us to write: 
```C#
T result = default; 
foreach (var x in source) { result = agg(result, x); } 
return result;
```

Reply:
we want a fold function without a seed, the initial default there doesn't work well with the usecase.

>would it be simpler to assume an aggregation function that does nothing on default 

that would only make the definition simpler, it has to be handled by the caller.